hoomd/GSDDequeWriter.cc

```diff
@@ -59,14 +59,42 @@ void GSDDequeWriter::analyze(uint64_t timestep)
         }
     }
 
-void GSDDequeWriter::dump()
+void GSDDequeWriter::dump(long int start, long int end, bool clear_entire_buffer)
     {
-    for (auto i {static_cast<long int>(m_frame_queue.size()) - 1}; i >= 0; --i)
+    auto buffer_length = static_cast<long int>(m_frame_queue.size());
+    if (end > buffer_length)
+        {
+        throw std::runtime_error("Burst.dump's end index is out of range.");
+        }
+    if (start < 0 || start > buffer_length)
+        {
+        throw std::runtime_error("Burst.dump's start index is out of range.");
+        }
+    long int iterator_start, iterator_end;
+    if (end < 0)
+        {
+        iterator_end = buffer_length - start;
+        iterator_start = 0;
+        }
+    else
+        {
+        iterator_end = buffer_length - start;
+        iterator_start = buffer_length - end;
+        }
+    for (auto i = iterator_end - 1; i >= iterator_start; --i)
         {
         write(m_frame_queue[i], m_log_queue[i]);
         }
```

Comment:
Does this not write the trajectory in reverse? I could be misunderstanding how write works

Reply:
See line 54 which is `m_log_queue.push_front(getLogData());`. Why this is `push_front` I do not know. But frames are added at the front of the queue and removed from the end.

Reply:
It does not. There are tests that specifically test if correct frames are dumped out. The reason why this works is because the dequeue is for some reason ordered in reverse (unsure why)...